integration/components/groups/GroupDetail.tsx

```diff
@@ -133,6 +137,40 @@ export default class GroupDetail extends React.Component<GroupDetailProps, Group
         return null;
     }
 
+    /**
+     * Verify the group ID the user typed is correct before making the actual delete call.
+     */
+    roatateGroupPrivateKey = () => {
+        IronWeb.group
+            .rotatePrivateKey(this.props.group.groupID)
+            .then((groupKeyUpdate) => {
+                logAction(`Group private key rotation was successful`);
+                this.setState({needsRotation: groupKeyUpdate.needsRotation});
+                this.getRotateGroupPrivateKeyIcon();
+            })
+            .catch((error: IronWeb.SDKError) => {
+                logAction(`Group rotation error: ${error.message}. Error Code: ${error.code}`, "error");
+            });
+    };
+
+    getRotateGroupPrivateKeyIcon() {
+        if (this.state.groupMeta.isAdmin) {
+            if (this.state.needsRotation) {
+                return (
+                    <FloatingActionButton onClick={this.roatateGroupPrivateKey} mini backgroundColor={red700}>
+                        <ImageRotateRight />
+                    </FloatingActionButton>
+                );
+            } else
+                return (
+                    <FloatingActionButton onClick={this.roatateGroupPrivateKey} mini backgroundColor={grey400}>
+                        <ImageRotateRight />
+                    </FloatingActionButton>
+                );
```

Comment:
The formatting here is weird because you're using brackets for the if, but not for the else. Instead I'd just remove the else and do

```
if(){
   return..
}
return ..
```